src/core/app.lua

```diff
@@ -283,18 +289,41 @@ function pace_breathing ()
    end
 end
 
+local function exceeded_ingress_packet_drops (app)
+   if app.ingress_packet_drops then
+      local threshold = ingress_packet_drops.last_value + ingress_packet_drops.threshold
+      local current = app:ingress_packet_drops()
+      local exceeded = current > threshold
+      if exceeded then
+         ingress_packet_drops.last_value = current
+      end
+      return exceeded
+   end
+end
+
+local jit_flush = (function ()
+   local last_trigger = 0
+   return function (app)
+      local now = now()
+      if now < last_trigger then return end
+      jit.flush()
+      last_trigger = now + ingress_packet_drops.wait
+   end
+end)()
```

Comment:
I would go ahead and smash the stateful logic together into one function, where we return true if we should call jit.flush(), taking into account both the threshold and the wait parameter.  (Incidentally we will probably want to expose the number of jit.flush() calls in a counter, in the future, and maybe the time of the last flush; pretty important operational data.)  WDYT?  Maybe these are bad ideas :)